pint_xarray/conversion.py

```diff
@@ -215,20 +250,64 @@ def convert_units_variable(variable, units):
     return new_obj
 
 
+def convert_units_index(index, index_vars, units):
+    if not isinstance(index, PintIndex):
+        raise ValueError("cannot convert non-quantified index")
+
+    converted_vars = {}
+    failed = {}
+    for name, var in index_vars.items():
+        unit = units.get(name)
+        try:
+            converted = convert_units_variable(var, unit)
+            converted_vars[name] = strip_units_variable(converted)
+        except (ValueError, pint.errors.PintTypeError) as e:
+            failed[name] = e
+
+    if failed:
+        # raise exception group
+        raise ValueError("failed to convert index variables:", failed)
+
+    # TODO: figure out how to pull out `options`
+    converted_index = index.index.from_variables(converted_vars, options={})
```

Comment:
@benbovy, do you have any idea about how to best do this? `from_variables` is a class method, so it won't preserve any options that were already on the index, even if we don't pass an empty `dict`.

Reply:
I'm afraid there is no easy way of converting units (i.e, new coordinate values + re-create the underlying index + wrap it as a new `PintIndex`) that would be completely agnostic of the underlying index type.

Actually I would rather drop the index and let the user re-create it manually. This is also better behavior for custom indexes that may be expensive to build (e.g., a KDTree).

For convenience we might also choose to auto-convert indexes for Xarray built-in index types only (and pass the options in a hard-coded way, if any... currently `PandasIndex` and `PandasMultiIndex` do not expose any option).